cpp/util/ThreadPool.hpp

```diff
@@ -143,6 +145,12 @@ class ThreadPool {
 
 #ifdef _WIN32
     key = TlsAlloc();
+    if (key == TLS_OUT_OF_INDEXES)
+    {
+      CXX_LOG_ERROR("Thread pool out of TLS index");
+      throw SnowflakeTransferException(TransferError::INTERNAL_ERROR,
+                                               "Out of TLS index in the thread pool");
+    }
 #else
     pthread_key_create(&key, NULL);
```

Comment:
Can you also do the check for Linux platform. pthread_key_create fails when the system has insufficient memory or if max keys is reached.

Reply:
Good point. I will add it.